File: Linnworks/src/javascript/Factory.js
var Factory = 
{
	GetResponse: function(ext, token, server, body) {
		var url = "";
		
		if(server)
		{
			url = server + "/api/" + ext;
		} else {
			url = "https://api.linnworks.net/api/" + ext;
		}
		
		var resp = "";
		
		$.ajax({
			url: url,
			type: 'post',
			data: body,
			headers: 
			{
				"Authorization" : token
			},
			dataType: 'application/x-www-form-urlencoded',
			complete: function (data) {
				resp = data.responseText;
			},
			async: false
		});	
		
		return JSON.parse(resp);
	}
}

File: Linnworks/src/net/LinnworksAPI/Factory.cs
using System;
using System.IO;
using System.Net;

public static class Factory
{
    public static string GetResponse(string Extension, string Body, Guid Token, string Server)
    {
        if (string.IsNullOrEmpty(Server))
        {
            Server = "https://api.linnworks.net/api/";
        }
        else
        {
            Server += "/api/";
        }

        HttpWebRequest Request = (HttpWebRequest)WebRequest.Create(Server + Extension);

        Request.ContentType = "application/x-www-form-urlencoded";
        Request.Method = "POST";

        if (Token != Guid.Empty)
        {
            Request.Headers.Add("Authorization", Token.ToString());
        }

        using (StreamWriter writer = new StreamWriter(Request.GetRequestStream()))
        {
            writer.Write(Body);
        }

        try
        {
            using (StreamReader reader = new StreamReader(Request.GetResponse().GetResponseStream()))
            {
                return reader.ReadToEnd();
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine(ex.Message);
            return "";
        }
    }
}